docs/rawget-type-operator.md

```diff
@@ -0,0 +1,58 @@
+# `rawget` type operator
+
+## Summary
+
+This RFC proposes the addition on one type operator, `rawget`, which can be used to look up a specific property of another type *without* invoking the `__index` metamethod.
+
+## Motivation
+
+There exists `index` type operator that allow developers to obtain a type of a property from classes / tables. If the a type is not found in the given class / table, the operator recursively indexes into the `__index` metamethod to continue searching for the property. Sometimes, this is could be an unwanted behavior. For example, given this code: 
+
+```luau
+local var1 = {
+  property = "hello"
+}
+
+local var2 = setmetatable({ }, { __index = seq })
+
+type doesExist = index<typeof(var2), "property"> -- reduces to `string`
+```
+The `index` type operator is used to obtain the type of `property` from a class `var1`. The operator continues to search recursively through the `__index` metamethod if the property is not initially found. This behavior can lead to unexpected results, as shown in the code snippet where the `__index` metamethod of `var2` points to `var1`, causing the `index` operator to return `string` for the `property` key even though `var2` itself does not have a property.
```

Comment:
Don't really understand why it's unexpected given that it exactly matches the indexing at runtime.